[GTK] Pass available input devices from UI process to web process for Interaction Media Features implementation

carlosgcampos opened the pull request:
#### 8649273e51fd9658ad0df9caea382b4550e7b4b7
<pre>
[GTK] Pass available input devices from UI process to web process for Interaction Media Features implementation
<a href="https://bugs.webkit.org/show_bug.cgi?id=291192">https://bugs.webkit.org/show_bug.cgi?id=291192</a>

Reviewed by Adrian Perez de Castro.

Using the same code added in 293276@main for WPE, but getting the devices using GTK API.

* Source/WebCore/platform/PlatformScreen.h:
(WebCore::screenHasTouchDevice):
(WebCore::screenIsTouchPrimaryInputDevice): Deleted.
* Source/WebCore/platform/gtk/PlatformScreenGtk.cpp:
(WebCore::screenHasTouchDevice):
(WebCore::screenIsTouchPrimaryInputDevice): Deleted.
* Source/WebKit/PlatformGTK.cmake:
* Source/WebKit/Scripts/webkit/messages.py:
(conditions_for_header):
* Source/WebKit/Shared/WebProcessCreationParameters.h:
* Source/WebKit/Shared/WebProcessCreationParameters.serialization.in:
* Source/WebKit/Shared/glib/AvailableInputDevices.h:
* Source/WebKit/Shared/glib/AvailableInputDevices.serialization.in:
* Source/WebKit/UIProcess/glib/WebProcessPoolGLib.cpp:
(WebKit::toAvailableInputDevices):
(WebKit::availableInputDevices):
(WebKit::seatDevicesChangedCallback):
(WebKit::WebProcessPool::platformInitialize):
(WebKit::WebProcessPool::platformInitializeWebProcess):
(WebKit::WebProcessPool::platformInvalidateContext):
* Source/WebKit/WebProcess/WebPage/glib/WebPageGLib.cpp:
(WebKit::WebPage::hoverSupportedByPrimaryPointingDevice const):
(WebKit::WebPage::hoverSupportedByAnyAvailablePointingDevice const):
(WebKit::WebPage::pointerCharacteristicsOfPrimaryPointingDevice const):
(WebKit::WebPage::pointerCharacteristicsOfAllAvailablePointingDevices const):
* Source/WebKit/WebProcess/WebPage/gtk/WebPageGtk.cpp:
(WebKit::WebPage::hoverSupportedByPrimaryPointingDevice const): Deleted.
(WebKit::WebPage::hoverSupportedByAnyAvailablePointingDevice const): Deleted.
(WebKit::WebPage::pointerCharacteristicsOfPrimaryPointingDevice const): Deleted.
(WebKit::WebPage::pointerCharacteristicsOfAllAvailablePointingDevices const): Deleted.
* Source/WebKit/WebProcess/WebPage/wpe/WebPageWPE.cpp:
(WebKit::WebPage::hoverSupportedByPrimaryPointingDevice const): Deleted.
(WebKit::WebPage::hoverSupportedByAnyAvailablePointingDevice const): Deleted.
(WebKit::WebPage::pointerCharacteristicsOfPrimaryPointingDevice const): Deleted.
(WebKit::WebPage::pointerCharacteristicsOfAllAvailablePointingDevices const): Deleted.
* Source/WebKit/WebProcess/WebProcess.h:
* Source/WebKit/WebProcess/WebProcess.messages.in:
* Source/WebKit/WebProcess/glib/WebProcessGLib.cpp:
(WebKit::WebProcess::platformInitializeWebProcess):
(WebKit::WebProcess::setScreenProperties):

Canonical link: <a href="https://commits.webkit.org/293352@main">https://commits.webkit.org/293352@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fbba77a346c88246325e43f53905b3df28967e2f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/98661 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/18294 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/8531 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/103785 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/49249 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/100706 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/18587 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/26745 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/75115 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/32262 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/101665 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/14110 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/89101 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/55472 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/98146 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/13892 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/7066 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/48631 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/83845 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/7144 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/106157 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/25751 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/18771 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/84088 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/26128 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/85302 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/83573 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/28213 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/5892 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/19461 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/15996 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/25709 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/30891 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/25527 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/28847 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/27102 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

spenap reviewed (commented):
LGTM! Good to see both ports using the same design!

aperezdc reviewed (approved):
Nice one, and it even removes more lines of code than it adds!

carlosgcampos committed:
[GTK] Pass available input devices from UI process to web process for Interaction Media Features implementation
https://bugs.webkit.org/show_bug.cgi?id=291192

Reviewed by Adrian Perez de Castro.

Using the same code added in 293276@main for WPE, but getting the devices using GTK API.

* Source/WebCore/platform/PlatformScreen.h:
(WebCore::screenHasTouchDevice):
(WebCore::screenIsTouchPrimaryInputDevice): Deleted.
* Source/WebCore/platform/gtk/PlatformScreenGtk.cpp:
(WebCore::screenHasTouchDevice):
(WebCore::screenIsTouchPrimaryInputDevice): Deleted.
* Source/WebKit/PlatformGTK.cmake:
* Source/WebKit/Scripts/webkit/messages.py:
(conditions_for_header):
* Source/WebKit/Shared/WebProcessCreationParameters.h:
* Source/WebKit/Shared/WebProcessCreationParameters.serialization.in:
* Source/WebKit/Shared/glib/AvailableInputDevices.h:
* Source/WebKit/Shared/glib/AvailableInputDevices.serialization.in:
* Source/WebKit/UIProcess/glib/WebProcessPoolGLib.cpp:
(WebKit::toAvailableInputDevices):
(WebKit::availableInputDevices):
(WebKit::seatDevicesChangedCallback):
(WebKit::WebProcessPool::platformInitialize):
(WebKit::WebProcessPool::platformInitializeWebProcess):
(WebKit::WebProcessPool::platformInvalidateContext):
* Source/WebKit/WebProcess/WebPage/glib/WebPageGLib.cpp:
(WebKit::WebPage::hoverSupportedByPrimaryPointingDevice const):
(WebKit::WebPage::hoverSupportedByAnyAvailablePointingDevice const):
(WebKit::WebPage::pointerCharacteristicsOfPrimaryPointingDevice const):
(WebKit::WebPage::pointerCharacteristicsOfAllAvailablePointingDevices const):
* Source/WebKit/WebProcess/WebPage/gtk/WebPageGtk.cpp:
(WebKit::WebPage::hoverSupportedByPrimaryPointingDevice const): Deleted.
(WebKit::WebPage::hoverSupportedByAnyAvailablePointingDevice const): Deleted.
(WebKit::WebPage::pointerCharacteristicsOfPrimaryPointingDevice const): Deleted.
(WebKit::WebPage::pointerCharacteristicsOfAllAvailablePointingDevices const): Deleted.
* Source/WebKit/WebProcess/WebPage/wpe/WebPageWPE.cpp:
(WebKit::WebPage::hoverSupportedByPrimaryPointingDevice const): Deleted.
(WebKit::WebPage::hoverSupportedByAnyAvailablePointingDevice const): Deleted.
(WebKit::WebPage::pointerCharacteristicsOfPrimaryPointingDevice const): Deleted.
(WebKit::WebPage::pointerCharacteristicsOfAllAvailablePointingDevices const): Deleted.
* Source/WebKit/WebProcess/WebProcess.h:
* Source/WebKit/WebProcess/WebProcess.messages.in:
* Source/WebKit/WebProcess/glib/WebProcessGLib.cpp:
(WebKit::WebProcess::platformInitializeWebProcess):
(WebKit::WebProcess::setScreenProperties):

Canonical link: https://commits.webkit.org/293352@main
